SA/design_concept/content/design/functional_requirements.typ

```diff
@@ -0,0 +1,61 @@
+= Functional Requirements
+The following section describes all use cases identified for the VisualFP application.
+
+== Actors
+VisualFP has two actors:
+
+#terms(
+    terms.item(
+        "Student User",
+        [
+            The student user is the main user of VisualFP and therefore the main influence on the visualization design.
+            The user wants to learn functional programming using VisualFP.
+            They want to do that by visually composing functions in a simple UI.
+            The UI should simplify the understanding of functional concepts that many beginners have difficulties with.
+        ]
+    ),
+    terms.item(
+        "Expert User",
+        [
+            The expert user is an experienced professional that wants to use VisualFP to help them to better understand their code.
+            For that, they want to be able to import their existing Haskell project into VisualFP.
+        ]
+    )
+)
+
+== Use Cases
+
+@use_case_diagram gives an overview of all identified use-cases.
+By default, the usage of the term "user" in the use case description refers to the role "student user".
+
+#figure(
+    [TODO: insert diagram],
+    caption: "Use Case Diagram"
+)<use_case_diagram>
+
+=== UC1 - Composition of simple function
+A user wants to compose a simple function using pre-defined functions and e.g. Integer parameters.
+
+=== UC2 - Function execution
+A user wants to execute their visually composed functions.
+
+=== UC3 - Composition of function using list comprehension
+A user wants to compose a function using list comprehension.
```

Comment:
Not sure whether we should really focus on list comprehension. It is just syntactic sugar of Haskell, after all

Reply:
I did think about visualizations for list comprehension for a few hours and it is indeed not the easiest feature to visualize. Especially when trying to support all possible cases.
So I'm inclined to drop it. Instead we could/should include the do notation as a use case.

Reply:
We could think about the do notation, but I think we will have lots of thinks to do with even just the core language features, let alone the syntactic extensions.

Reply:
I'll rewrite the use-case to general list handling